refactor (sidebar>layout.tsx):

1MeghPatel1 opened the pull request:
- moved the client side interactivity down the component tree.
- refatored LeftSideBar componet for cleaner code.
- made a separate CreatePostFormButton and LeftSideBarNavLinks components.
- added loading.tsx file for (with-sidebar) route group for showing loader for page navigations.

1MeghPatel1 committed:
refactor (sidebar>layout.tsx):
- moved the client side interactivity down the component tree
- refatored LeftSideBar componet for cleaner code
- made a separate CreateFormButton and LeftSideBarNavLinks components

1MeghPatel1 committed:
fix (CreatePostFormButton):
- fixed file name to avoid duplicate names

thekalamanchofficial commented on src/app/(with-sidebar)/myfeed/types/types.ts:
```diff
@@ -123,3 +123,11 @@ export type FeaturedPost = {
   likeCount: number;
 };
+
+export type LeftSideBarNavLinksProps = {
```
move the prop type to component itself, and declare the type for menu items here

1MeghPatel1 replied:
Done, moved it to LeftSideBarNavLinks.tsx

thekalamanchofficial replied:
You did not declare the type for menuItems, that should be in the types file

thekalamanchofficial commented on src/app/_components/sidebar/CreatePostFormButton.tsx:
```diff
@@ -0,0 +1,68 @@
+  };
+
+  return (
+    <>
```
remove the fragment

1MeghPatel1 replied:
Done, removed fragment

1MeghPatel1 replied:
Also, formatted code

thekalamanchofficial commented on src/app/_components/sidebar/LeftSideBar.tsx:
```diff
@@ -1,161 +1,45 @@
+import { STATIC_TEXTS } from "~/app/_components/static/staticText";
+import CreatePostFormButton from "./CreatePostFormButton";
 import UserMenu from "../userMenu/UserMenu";
+import { type LeftSideBarProps } from "~/app/(with-sidebar)/myfeed/types/types";
```
declare the prop type in same file

thekalamanchofficial commented on src/app/_components/sidebar/LeftSideBar.tsx:
```diff
@@ -1,161 +1,45 @@
-          const isActive = pathname === item.route;
+      <Divider sx={{ width: "100%" }} />
+      
+      {/* Create Post Form Button */}
```
dont need this comment

1MeghPatel1 replied:
Done, removed comment

thekalamanchofficial commented on src/app/_components/sidebar/LeftSideBar.tsx:
```diff
@@ -1,161 +1,45 @@
-          );
-        })}
+      <Box sx={{ marginTop: 2, px: "8px" }}>
+        {/* Pass menuItems to the client-side NavLink component */}
```
dont need the comment

1MeghPatel1 replied:
Done, removed comment

1MeghPatel1 committed:
fix (LeftSideBar.tsx):
- removed unnecessary comments
- removed react fragment from CreatePostFormButton

1MeghPatel1 committed:
fix (LeftSideBar.tsx):
- moved the prop types to its component
